src/System.Text.Json/src/System/Text/Json/Node/JsonNumber.cs

```diff
@@ -0,0 +1,664 @@
+        /// <exception cref="ArgumentNullException">
+        ///   Provided value is null.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   Provided value is not in a legal JSON number format.
+        /// </exception>
+        /// <remarks>Provided value is stored in the same format as passed.</remarks> 
+        public void SetFormattedValue(string value)
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                if (value == null)
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+                throw new ArgumentException(SR.EmptyStringToInitializeNumber, nameof(value));
+            }              
+            
+            JsonWriterHelper.ValidateNumber(Encoding.UTF8.GetBytes(value).AsSpan());
+            _value = value;
+        }
+
+        /// <summary>
+        ///   Changes the numeric value of this instance to represent a specified <see cref="byte"/> value.
+        /// </summary>
+        /// <param name="value">The value to represent as a JSON number.</param>
+        public void SetByte(byte value) => _value = value.ToString();
+
+        /// <summary>
+        ///   Changes the numeric value of this instance to represent a specified <see cref="short"/> value.
+        /// </summary>
+        /// <param name="value">The value to represent as a JSON number.</param>
+        public void SetInt16(short value) => _value = value.ToString();
+
+        /// <summary>
+        ///   Changes the numeric value of this instance to represent a specified <see cref="int"/> value.
+        /// </summary>
+        /// <param name="value">The value to represent as a JSON number.</param>
+        public void SetInt32(int value) => _value = value.ToString();
+
+        /// <summary>
+        ///   Changes the numeric value of this instance to represent a specified <see cref="long"/> value.
+        /// </summary>
+        /// <param name="value">The value to represent as a JSON number.</param>
+        public void SetInt64(long value) => _value = value.ToString();
+
+        /// <summary>
+        ///   Changes the numeric value of this instance to represent a specified <see cref="float"/> value.
+        /// </summary>
+        /// <param name="value">The value to represent as a JSON number.</param>
+        public void SetSingle(float value)
+        {
+            if (value.Equals(float.PositiveInfinity) || value.Equals(float.NegativeInfinity))
+            {
+                throw new FormatException(SR.IllegalNumberFormat);
+            }
+            _value = value.ToString();
+        }
+        /// <summary>
+        ///   Changes the numeric value of this instance to represent a specified <see cref="double"/> value.
+        /// </summary>
+        /// <param name="value">The value to represent as a JSON number.</param>
```

Comment:
Add exception xml tag in the comments to document what exceptions are thrown and when.